File: Java/Maximum_SubArray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        
        
        int max_final=nums[0];
        int max_till_here=0;
        for(int i=0;i<nums.length;i++)
        {
            //max_till_here=max_till_here+nums[i];
            max_till_here=Math.max(max_till_here+nums[i],nums[i]);
            
            max_final = Math.max(max_final,max_till_here);
            /*if(max_till_here<0) //if the sum is less than zero than reset(another approach)
            {
                max_till_here=0;
            }*/
        }
        return max_final;
    }
}

File: Python/MaximumSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_final=nums[0]
        max_till_here=0
        
        for num in nums:
            max_till_here=max(num,max_till_here+num)
            max_final=max(max_final,max_till_here)
        
        return max_final